apps/products/models.py

```diff
@@ -0,0 +1,250 @@
+    def delete_branches_picture(sender, instance, **kwargs):
+        if instance.picture and instance.picture.name:
+            file_path = instance.picture.path
+            if os.path.exists(file_path):
+                os.remove(file_path)
+                picture_dir = os.path.dirname(file_path)
+                if not os.listdir(picture_dir):
+                    os.rmdir(picture_dir)
+
+
+
+class ProductStockUnits(models.Model):
+    stock_unit_id = models.AutoField(primary_key=True)
+    stock_unit_name = models.CharField(max_length=255)
+    description = models.TextField()
+    quantity_code_id = models.ForeignKey(ProductUniqueQuantityCodes, on_delete=models.CASCADE, null=True, default=None, db_column = 'quantity_code_id')
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    class Meta:
+        db_table = productstockunitstable
+
+    def __str__(self):
+        return f"{self.stock_unit_id} {self.stock_unit_name}"
+
+
+class ProductGstClassifications(models.Model):
+    TYPE_CHOICES = [ 
+        ('HSN', 'HSN'),
+        ('SAC', 'SAC'),
+        ('Both','Both'),
+    ]
+    gst_classification_id = models.AutoField(primary_key=True)
+    type = models.CharField(max_length=10, choices=TYPE_CHOICES, blank=True, null=True)
+    code = models.CharField(max_length=50)
+    hsn_or_sac_code = models.CharField(max_length=50)
+    hsn_description = models.TextField()
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    class Meta:
+        db_table = productgstclassificationstable
+
+    def __str__(self):
+        return f"{self.gst_classification_id} {self.code}"
+
+
+class ProductSalesGl(models.Model):
+    sales_gl_id = models.AutoField(primary_key=True)
+    ledger_group_id = models.ForeignKey(LedgerGroups, on_delete=models.CASCADE, null=True, default=None, db_column = 'ledger_group_id')
+    name = models.CharField(max_length=255)
+    sales_accounts = models.CharField(max_length=255)
+    code = models.CharField(max_length=50)
+    is_subledger = models.BooleanField(default=False)
+    inactive = models.BooleanField(default=False)
+    type = models.CharField(max_length=255)
+    account_no = EncryptedTextField(max_length=255)
+    rtgs_ifsc_code = models.CharField(max_length=255)
+    classification = models.CharField(max_length=255)
+    is_loan_account = models.BooleanField(default=False)
+    tds_applicable = models.BooleanField(default=False)
+    address = models.CharField(max_length=255)
+    pan = models.CharField(max_length=50)
+    employee = models.BooleanField(default=False)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    class Meta:
+        db_table = 'product_sales_gl'
+
+    def __str__(self):
+        return f"{self.sales_gl_id} {self.name}"
+
+class ProductPurchaseGl(models.Model):
```

Comment:
Why this class is duplicated?

Reply:
No duplicate class is created Thulasi 
ProductSalesGl
ProductPurchaseGl
both are different classes Thulasi